src/components/Icon/index.js

```diff
@@ -1,29 +1,38 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 
 function Icon({
   className,
   family,
   type,
+  hoverIconType,
```

Comment:
I would remove the `Icon` part from the name. IMHO it is a bit redundant. Having the attributes `type` and `hoverType` seems slightly better than `type` and `hoverIconType`.